### ISSUE TITLE
#det and #det_exact disagree

### ISSUE THREAD

**Issue by agisga** [SciRuby/nmatrix#368]
When working with NMatrix today, I noticed the following strange behaviour:

```Ruby
m = NMatrix.new([2,2], [1,2,3,4])  # =>[  [1, 2]   [3, 4] ]
m.det             # => 2
m.det_exact   # => -2
```

The correct determinant should be -2, as computed by `#det_exact` but not by `#det`. Am I missing something here?

**Comment by mohawkjohn** [comment]
Oy. Definitely a bug. #334, #347 may be related.

**Comment by wlevine** [comment]
det calls getrf, so I suspect this might be due to #360

**Comment by mohawkjohn** [comment]
Oops. In that case, I'm going to mark this as a duplicate of #360 and close it. Thanks, @wlevine.

**Comment by mohawkjohn** [comment]
Fixed by #370
